tsl/test/expected/compression_update_delete.out

```diff
@@ -1974,3 +1974,110 @@ SELECT * FROM chunk_status;
 (4 rows)
 
 ROLLBACK;
+DROP TABLE join_test1;
+DROP TABLE join_test2;
+-- test if index scan qualifiers are properly used
+CREATE TABLE index_scan_test(time timestamptz NOT NULL, device_id int, value float);
+SELECT create_hypertable('index_scan_test','time',create_default_indexes:=false);
+       create_hypertable       
+-------------------------------
+ (29,public,index_scan_test,t)
+(1 row)
+
+INSERT INTO index_scan_test(time,device_id,value) SELECT time, device_id, device_id + 0.5 FROM generate_series('2000-01-01 0:00:00+0'::timestamptz,'2000-01-01 23:55:00+0','1m') gtime(time), generate_series(1,5,1) gdevice(device_id);
+-- compress chunks
+ALTER TABLE index_scan_test SET (timescaledb.compress, timescaledb.compress_orderby='time DESC', timescaledb.compress_segmentby='device_id');
+SELECT compress_chunk(show_chunks('index_scan_test'));
+              compress_chunk              
+------------------------------------------
+ _timescaledb_internal._hyper_29_59_chunk
+(1 row)
+
+ANALYZE index_scan_test;
+SELECT ch1.schema_name|| '.' || ch1.table_name AS "CHUNK_1"
+FROM _timescaledb_catalog.chunk ch1, _timescaledb_catalog.hypertable ht
+WHERE ht.table_name = 'index_scan_test'
+AND ch1.hypertable_id = ht.id
+AND ch1.table_name LIKE '_hyper%'
+ORDER BY ch1.id LIMIT 1 \gset
+SELECT ch2.schema_name|| '.' || ch2.table_name AS "COMP_CHUNK_1"
+FROM _timescaledb_catalog.chunk ch1, _timescaledb_catalog.chunk ch2, _timescaledb_catalog.hypertable ht
+WHERE ht.table_name = 'index_scan_test'
+AND ch1.hypertable_id = ht.id
+AND ch1.compressed_chunk_id  = ch2.id
+ORDER BY ch2.id LIMIT 1 \gset
+INSERT INTO index_scan_test(time,device_id,value) SELECT time, device_id, device_id + 0.5 FROM generate_series('2000-01-01 0:00:00+0'::timestamptz,'2000-01-05 23:55:00+0','1m') gtime(time), generate_series(1,5,1) gdevice(device_id);
+BEGIN;
+SELECT count(*) as "UNCOMP_LEFTOVER" FROM ONLY :CHUNK_1 WHERE device_id != 2 \gset
+CREATE INDEX ON index_scan_test(device_id);
+EXPLAIN (costs off, verbose) DELETE FROM index_scan_test WHERE device_id = 2;
+                                                                 QUERY PLAN                                                                  
+---------------------------------------------------------------------------------------------------------------------------------------------
+ Custom Scan (HypertableModify)
+   ->  Delete on public.index_scan_test
+         Delete on _timescaledb_internal._hyper_29_59_chunk index_scan_test_1
+         ->  Index Scan using _hyper_29_59_chunk_index_scan_test_device_id_idx on _timescaledb_internal._hyper_29_59_chunk index_scan_test_1
+               Output: index_scan_test_1.tableoid, index_scan_test_1.ctid
+               Index Cond: (index_scan_test_1.device_id = 2)
```

Comment:
Question:
Attribute no: of device_id in _hyper_29_59_chunk is 2.
Attribute no: of device_id in _hyper_29_59_chunk_index_scan_test_device_id_idx is 1.

Ideally since index condition is on index, the attribute no: of device id should be 1. Right ?
Instead i see that its 2, thus the followup code path works as expected.

Want to understand why columns names in Index cond qualifiers use attribute no: of chunks.

Reply:
indexqualorig is the original qualifier on the table while indexqual member in the IndexScan uses the actual index qualifiers.

Reply:
Cool, thanks for the clarification.